tasks/bash.sh

```diff
@@ -25,18 +25,37 @@ success() {
   echo "$1"
 }
 
+determine_command_for_facter_4() {
+  puppet_version="$(puppet --version)"
+
+  if (( ${puppet_version%%.*} == 6 )); then
+    # puppet 6 with facter 4
+    facts_command="facter --json --show-legacy"
```

Comment:
This still regresses in that it doesn't show Puppet facts. It seems like `puppet facts show` will still work as expected with Puppet 6, so we should use that whenever we detect Facter 4 regardless of the Puppet version.

Reply:
`puppet facts show` will only work starting with puppet 7. In puppet 6 if you run `puppet facts show`, it will translate this to `puppet facts find show` e.g.
```
puppet --version
6.19.0
```

```
{
  "name": "show",
  "values": {
    "os": {
      "distro": {
        "release": {
          "full": "20.04",
          "major": "20.04"
        },
        "codename": "focal",
        "description": "Ubuntu 20.04 LTS",
        "id": "Ubuntu"
      },
      "release": {
        "full": "20.04",
        "major": "20.04"
      },
      "architecture": "amd64",
      "family": "Debian",
      "hardware": "x86_64",
      "selinux": {
        "enabled": false
      },
      "name": "Ubuntu"
    },
....
```

`puppet facts find` does not know to query after a specific fact and it will append the `show` string to it's results
`"name": "show",`

Reply:
With puppet 7
```
puppet --version
7.0.0
```

```
puppet facts show
{
   "os":{
      "distro":{
         "release":{
            "full":"20.04",
            "major":"20.04"
         },
         "codename":"focal",
         "id":"Ubuntu",
         "description":"Ubuntu 20.04 LTS"
      },
      ...
```

in this case `puppet facts show` is the real command, this is NOT translated to `puppet facts find show` and the result is identical to what was previously obtained with `facter -p`

Reply:
Ah ok, I see